File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int currSum = 0;

        for(int n: nums){
            currSum = Math.max(currSum+n, n);
            maxSum = Math.max(maxSum, currSum);
        }
        return maxSum;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.go
func findMaxConsecutiveOnes(nums []int) int {
	L := 0
	R := 0
	max := 0

	for R < len(nums) && len(nums) - L >= max {
		if nums[R] == 1 {
			R++
		}else{
			tmp := R - L
			if tmp > max {
				max = tmp
			}
			R++
			L = R
		}
	}

	if R - L > max {
		return R-L
	}

	return max
	
}
